x/mint/minter.go

```diff
@@ -7,45 +7,57 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// current inflation state
+// Minter represents the minting state
 type Minter struct {
-	InflationLastTime time.Time `json:"inflation_last_time"` // block time which the last inflation was processed
-	Inflation         sdk.Dec   `json:"inflation"`           // current annual inflation rate
+	LastUpdate       time.Time `json:"last_update"`       // time which the last update was made to the minter
+	Inflation        sdk.Dec   `json:"inflation"`         // current annual inflation rate
+	AnnualProvisions sdk.Int   `json:"annual_provisions"` // current annual exptected provisions
 }
 
-// minter object for a new minter
-func InitialMinter() Minter {
+// Create a new minter object
+func NewMinter(lastUpdate time.Time, inflation sdk.Dec,
+	annualProvisions sdk.Int) Minter {
+
 	return Minter{
-		InflationLastTime: time.Unix(0, 0),
-		Inflation:         sdk.NewDecWithPrec(13, 2),
+		LastUpdate:       lastUpdate,
+		Inflation:        inflation,
+		AnnualProvisions: annualProvisions,
 	}
 }
 
+// minter object for a new chain
+func InitialMinter(inflation sdk.Dec) Minter {
+	return NewMinter(
+		time.Unix(0, 0),
+		inflation,
+		sdk.NewInt(0),
+	)
+}
+
+// default initial minter object for a new chain
+func DefaultInitialMinter() Minter {
+	return InitialMinter(
+		sdk.NewDecWithPrec(13, 2),
+	)
+}
+
 func validateMinter(minter Minter) error {
 	if minter.Inflation.LT(sdk.ZeroDec()) {
-		return fmt.Errorf("mint parameter Inflation should be positive, is %s ", minter.Inflation.String())
+		return fmt.Errorf("mint parameter Inflation should be positive, is %s",
+			minter.Inflation.String())
 	}
 	if minter.Inflation.GT(sdk.OneDec()) {
```

Comment:
Why don't we allow inflations above a hundred percent? We (or other SDK chains) might want > 100% inflation.

Reply:
you're right, I'll remove this here now